File: ndk/NativeLog/app/src/main/java/org/example/nativelog/MainActivity.java
package org.example.nativelog;

import android.support.v7.app.ActionBarActivity;
import android.os.Bundle;

public class MainActivity extends ActionBarActivity {

	@Override protected void onCreate(Bundle savedInstanceState) {
		super.onCreate(savedInstanceState);
		setContentView(R.layout.activity_main);
		main();
	}

	public native void main();

	static {
		System.loadLibrary("scene");
	}
}

File: kotlin/SeekBar/app/src/main/java/example/kotlin/seekbar/MainActivity.kt
package example.kotlin.seekbar

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.widget.SeekBar
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

	override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		setContentView(R.layout.activity_main)

		seekBar.setOnSeekBarChangeListener(object : SeekBar.OnSeekBarChangeListener {
			override fun onProgressChanged(seekBar: SeekBar?, progress: Int, fromUser: Boolean) {
			}

			override fun onStartTrackingTouch(seekBar: SeekBar?) {
			}

			override fun onStopTrackingTouch(seekBar: SeekBar?) {
				Toast.makeText(this@MainActivity, "Progress is ${seekBar?.progress}%", Toast.LENGTH_SHORT).show()
			}
		})
	}
}
